src/Microsoft.AspNet.Authentication.Cookies/CookieAuthenticationOptions.cs

```diff
@@ -152,5 +153,13 @@ public string CookieName
         /// to the client. This can be used to mitigate potential problems with very large identities.
         /// </summary>
         public IAuthenticationSessionStore SessionStore { get; set; }
+
+        CookieAuthenticationOptions IOptions<CookieAuthenticationOptions>.Value
```

Comment:
I tried, no good, abstract base class messes with all the inheritance chains.  It was really messy and ended up adding `AuthenticationOptions<TOptions>` : `Options<TOptions>` and it exploded into all the base options...  My conclusion NOT worth it